ddtrace/settings/dynamic_instrumentation.py

```diff
@@ -0,0 +1,79 @@
+from envier import En
+
+from ddtrace import config
+from ddtrace.internal.agent import get_trace_url
+from ddtrace.internal.constants import DEFAULT_SERVICE_NAME
+from ddtrace.internal.utils.config import get_application_name
+from ddtrace.version import get_version
+
+
+DEFAULT_MAX_PROBES = 100
+DEFAULT_GLOBAL_RATE_LIMIT = 100.0
+
+
+def derive_tags(c):
+    # type: (En) -> str
+    _tags = dict(env=config.env, version=config.version, debugger_version=get_version())
+    _tags.update(config.tags)
+
+    return ",".join([":".join((k, v)) for (k, v) in _tags.items() if v is not None])
+
+
+class DynamicInstrumentationConfig(En):
```

Comment:
Do we want this to be public?

Reply:
Well, in a sense this is. It's just a "parsed" version of the environment, no?

Reply:
The concern is more if we'd want to change this data structure and whether it's useful for a user. I don't think a user would have use of it so just easier to leave private in case we want to change it in the future

Reply:
I see your point, but I think this is unlikely to change, at least as far as the public attributes are concerned. If they do, they will have the same deprecation warning and period as the corresponding environment variable.